basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/HostTestCase.java

```diff
@@ -129,12 +174,44 @@ public void addHTTPServerHostInGUI() throws Exception {
 
     @Test
     public void removeHTTPServerHostInGUI() throws Exception {
-        ConfigFragment config = page.getConfigFragment();
-        config.getResourceManager()
+        page.getResourceManager()
                 .removeResource(HOST_TBR)
                 .confirm();
 
-        Assert.assertFalse("Host server host should not be present in table", config.resourceIsPresent(HOST_TBR));
+        Assert.assertFalse("Host server host should not be present in table", page.getResourceManager().isResourcePresent(HOST_TBR));
         new ResourceVerifier(HOST_TBR_ADDRESS, client).verifyDoesNotExist(); //HTTP server host should not be present on the server
     }
+
+    @Test
+    public void checkReferencesToFilters() throws IOException {
+        page.switchToReferenceToFilterSubTab();
+
+        Assert.assertTrue("Filter reference " + ERROR_PAGE_FILTER_NAME + " is not present in table!",
+                page.getConfigFragment().getResourceManager().isResourcePresent(ERROR_PAGE_FILTER_NAME));
+
+        Assert.assertTrue("Filter reference  " + REQUEST_LIMIT_FILTER_NAME + " is not present in table!",
+                page.getConfigFragment().getResourceManager().isResourcePresent(REQUEST_LIMIT_FILTER_NAME));
+    }
+
+    @Test
+    public void addReferenceToFilter() throws IOException {
+        page.switchToReferenceToFilterSubTab();
+        page.addReferenceToFilter(GZIP_PAGE_FILTER_NAME);
+
+        Assert.assertTrue("Filter reference  " + GZIP_PAGE_FILTER_NAME + " is not present in table!",
+                page.getConfigFragment().getResourceManager().isResourcePresent(GZIP_PAGE_FILTER_NAME));
```

Comment:
You should also verify that it was added to model and not that it is just updated in UI